go/processor/floria/run_context.go

```diff
@@ -61,10 +61,14 @@ func (r runContext) executeCall(kind tosca.CallKind, parameters tosca.CallParame
 		r.static = true
 	}
 
-	if r.blockParameters.Revision >= tosca.R09_Berlin &&
-		!isPrecompiled(recipient, r.blockParameters.Revision) &&
-		!isStateContract(recipient) &&
-		!r.AccountExists(recipient) &&
+	isStateContract := isStateContract(parameters.CodeAddress)
+	isPrecompiled := isPrecompiled(parameters.CodeAddress, r.blockParameters.Revision)
+
+	if kind == tosca.Call &&
+		r.blockParameters.Revision >= tosca.R09_Berlin &&
+		!isPrecompiled &&
+		!isStateContract &&
+		!r.AccountExists(parameters.Recipient) &&
 		parameters.Value.Cmp(tosca.Value{}) == 0 {
 		return tosca.CallResult{Success: true, GasLeft: parameters.Gas}, nil
 	}
```

Comment:
This looks like an optimization of a call into a non-existing account. Can this be removed without changing anything?